### PR TITLE
[MAT-7156, MAT-7162] Shift QICore Test Case Dates

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-7156](https://jira.cms.gov/browse/MAT-7156), [MAT-7162](https://jira.cms.gov/browse/MAT-7162)
(Optional) Related Tickets:

### Summary

Move canned test files. Old directory didn't make sense, as all the files were not part of different measures,
Add Test Case date shift capability (developed by Ashok, See MAT-7433).
Test leap year handling.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+@Tag(name = "Test-Case-Controller", description = "API for manipulating FHIR Test Cases")
+@RequiredArgsConstructor
+public class TestCaseController {
+  private final TestCaseDateShifterService testCaseDateShifterService;
+
+  @PutMapping("/test-cases/shift-dates")
+  public ResponseEntity<List<TestCase>> shiftTestCasesDates(
+      Principal principal,
+      @RequestBody List<TestCase> testCases,
+      @RequestParam(name = "shifted", defaultValue = "0") int shifted) {
```
Comment (sb-prateekkeerthi) [inline src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java:28]: Suggestion: shouldn't the name be shift as we are shifting it now. Shifted sounds past tense

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java:28]: Makes sense. I was trying to be consistent with the existing QDM date shift methods in measure-service so that we didn't have differing parameters between the services.

Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java:28]: Okay. I'll approve it then, rest looks good to me

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java:28]: How about "years"? The api already has `shfit-dates` so the parameter could document the unit.

e.g. `/test-cases/shift-dates?years=X`

**Review by sb-prateekkeerthi — DISMISSED** [review]

File: src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/madiefhirservice/resources/TestCaseController.java:28]: yes that would work

File: src/main/java/gov/cms/madie/madiefhirservice/services/TestCaseDateShifterService.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    for (Field field : fields) {
+      Property property = baseResource.getNamedProperty(field.getName());
+      if (property != null) {
+        List<Base> values = property.getValues();
+        for (Base value : values) {
+          if (value.isPrimitive()) {
+            if (value.isDateTime()) {
+              BaseDateTimeType dateType = (BaseDateTimeType) value;
+              dateType.add(1, shifted);
+              baseResource.getNamedProperty(field.getName());
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/TestCaseDateShifterService.java]: this might be some testing/debugging thing

Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/madiefhirservice/services/TestCaseDateShifterService.java]: Ah, I see now, that last line isn't doing anything meaningful. Let me remove it.

**Review by adongare — APPROVED** [review]
